inginious/frontend/pages/course_custom_values.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+import flask
+from inginious.frontend.pages.utils import INGIniousAuthPage
+
+
+class CustomValuePage(INGIniousAuthPage):
+    def GET_AUTH(self, courseid):
+        """ GET request """
+        username = self.user_manager.session_username()
+        current_user = self.database.users.find_one(
+            {"username": username})
+        custom_values_user = current_user["custom"] if "custom" in current_user else {}
+        return self.show_page(courseid, custom_values_user)
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        user_input = flask.request.form
+        username = self.user_manager.session_username()
+        current_user = self.database.users.find_one(
+            {"username": username})
+        custom_values_user = current_user["custom"] if "custom" in current_user else {}
+        for key in user_input:
+            custom_values_user[key] = user_input[key]
+        removed_keys = list(set(custom_values_user.keys()) - set(user_input.keys()))
+        for key in removed_keys:
+            custom_values_user[key] = "off"
+        self.database.users.update_one({"username": username}, {"$set": {"custom": custom_values_user}})
```

Comment:
All those fields are defined for the user/course context. You should add a level to this dictionary not to conflict with other courses...